### ISSUE TITLE
Does this library work with Non-boot Spring projects?

### ISSUE THREAD

**Issue by nzall** [springdoc/springdoc-openapi#568]
We have a Spring REST API that we want to automatically generate documentation for. We are currently using Springfox in our development environment (no release version has autogenerated documentation yet), but are investigating migrating to Springdoc because Springfox doesn't seem to be going anywhere with Java 11 compatibility. However, our project does not use Spring-Boot. Has this library been tested yet with non-Spring Boot projects?

**Comment by bnasslahsen** [comment]
@nzall, 

 The library integrates natively with spring-boot.
 If you want to use spring, its possible but the integration details will not be covered as it really depend on the version/module and the nature of you spring application.

 You can have a look here #250, where there is the steps for some kind of application and sample code using spring without spring-boot.

**Comment by nzall** [comment]
@bnasslahsen Thank you for the quick response.

You mentioned in #250 that you will update the FAQ and Stack Overflow. I might be missing the obvious here and my apologies if I do, but I do not see any entry in the FAQ table of contents about using springdoc with a a library without spring boot or anything that looks related. I haven't read the full FAQ yet, but before I fully read through it, could you tell me if this promised FAQ update has been completed yet?

**Comment by bnasslahsen** [comment]
@nzall 

Here is the link for the F.A.Q:
- https://springdoc.org/faq.html#how-to-integrate-open-api-3-with-spring-project-not-spring-boot

As mentionned before, there is no real guide for spring integration.  It will give you an idea of the steps to follow.
It really depends on your application context and on the version of spring are using.

**Comment by lrozenblyum** [comment]
@bnasslahsen Recently the visibility of configuration classes was reduced
https://github.com/springdoc/springdoc-openapi/commit/00b2e02cc41ecb5f3d819d8df28cc289f6c0a3ce
so it's impossible to register part of those classes as described in the FAQ

**Comment by bnasslahsen** [comment]
@lrozenblyum, 

 I see and understand.
 Will be reverted then for this usage for v1.3.7.

**Comment by lrozenblyum** [comment]
Thanks @bnasslahsen !

**Comment by bnasslahsen** [comment]
@lrozenblyum, 

 Its now available on master. 
 If you see any missing class. Don't hesistate to let me know before the next release.

**Comment by lrozenblyum** [comment]
Thanks @bnasslahsen it looks like all the needed classes are public again.

**Comment by gurmindersing** [comment]
We have a spring webflux reactive application that needs to be deployed as war on server. The spring boot cannot be used as dependency as spring boot does not support spring webflux app deployed as war.

As per solution mention in github ticket to make springdoc work, do we need to add spring boot dependencies?

Is springdoc compatible with non spring boot webflux application?

**Comment by BhargavRuparel3591** [comment]
We have Spring framework (version 5.3.22) based application, not a spring boot.

We have added springdoc-openapi (1.6.11) dependency along with below openapi configuration.

@Configuration 
@EnableWebMvc 
@ComponentScan(basePackages = {"org.springdoc","rest"}) 
@Import({org.springdoc.core.SpringDocConfiguration.class, 
         org.springdoc.webmvc.core.SpringDocWebMvcConfiguration.class, 
         org.springdoc.webmvc.ui.SwaggerConfig.class, 
         org.springdoc.core.SwaggerUiConfigProperties.class, 
         org.springdoc.core.SwaggerUiOAuthProperties.class, 
         org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration.class}) 
public class OpenApiConfig implements WebMvcConfigurer { }


In our application, we have multiple projects having mixed of Java configurations and XML based configurations.

Both types of classes (java config like @controller/@service and XML definitions like <bean id="".... >) are having dependencies injected through @Autowred.

The classes having beans created through component scanning (@controller/@service) are working well, all the injected dependecies are resolved as expected.
But the classes having XML definitions are having issues, where dependencies injected through setter and bean ref are working fine, but the dependencies injected through @Autowired are found as null at runtime. Looks like @Autowired annotation itself is not getting resolved while creating bean of such classes. We have also verified that our XML context file has <context:annotation-config /> defined and so @Autowired should get resolved here.

Without this springdoc-openpi dependency and these OpenApiConfig, we are not facing this issue.

Does anyone have more insights here ?
